From ab047ba47f933f48cf915383e943a51391b6bdff Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Wed, 27 Apr 2016 18:39:17 +0530
Subject: [PATCH 4/4] EM/dc_xpwr_charger: Conditionally register vbus control
 cooling device

Issue:
VBUS control is a cooling device register to control VBUS drive to manage
proactive peak current management. Typically when the battery level is low,
user can control the consumption by disabling the USB devices connected to
the DUT. Current implementation in dollar cove charger driver registers for
a vbus control no matter for which device the GPIO resource for controlling
the VBUS is linked to. This caused issues if this VBU GPIO is not defined
for PMIC device. OTG2 device (INT3496 driver) typically implements this
GPIO control. In that case, this vbus control registry will not have any
effect.

Solution:
Register vbus control only if the GPIO resource is defined for PMIC device
in BIOS. Else do not register for the vbus control in this driver.

Change-Id: I7593fc4eecfdfa2c180dd87e6d1dc4c1ec312749
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30033
Reviewed-on: https://android.intel.com:443/500739
---
 drivers/power/dc_xpwr_charger.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/drivers/power/dc_xpwr_charger.c b/drivers/power/dc_xpwr_charger.c
index fc13899..b385007 100644
--- a/drivers/power/dc_xpwr_charger.c
+++ b/drivers/power/dc_xpwr_charger.c
@@ -1037,15 +1037,16 @@ static int pmic_chrg_probe(struct platform_device *pdev)
 			 * on usb id during boot.
 			 */
 			schedule_work(&info->otg_work);
-		}
-	}
 
-	/* Register cooling device to control the vbus */
-	ret = register_cooling_device(info);
-	if (ret) {
-		dev_err(&info->pdev->dev,
-			"Register cooling device Failed (%d)\n", ret);
-		goto cdev_reg_fail;
+			/* Register cooling device to control the vbus */
+			ret = register_cooling_device(info);
+			if (ret) {
+				dev_err(&info->pdev->dev,
+					"Register cooling device Failed (%d)\n",
+					ret);
+				goto cdev_reg_fail;
+			}
+		}
 	}
 
 	return 0;
-- 
1.9.1

